pkg/plugins/resources/file/main_test.go

```diff
@@ -103,77 +149,115 @@ func Test_Validate(t *testing.T) {
 
 func TestFile_Read(t *testing.T) {
 	tests := []struct {
-		name                string
-		wantErr             bool
-		mockReturnedContent string
-		mockReturnedError   error
-		spec                Spec
-		wantContent         string
-		wantMockState       text.MockTextRetriever
+		name           string
+		spec           Spec
+		files          map[string]string
+		mockedContents map[string]string
+		mockedLines    map[string]int
+		mockedError    error
+		wantedContents map[string]string
+		wantedLines    map[string]int
```

Comment:
Not sure about the pertinence of `mockedLines` and `wantedLines` now, they don't add anything IMO and should be removed.

Reply:
Haven't check yet: weren't some source and condition tests using it?
(but I feel lik that your improvement of the mock might have had the possibility to make these attributes obsolete, and god knows how I love code being removed \o/)

Reply:
Some tests include them but I don't see the added values since the mock now returns and writes with the specified line parameter

Reply:
There are no changes between the initial mocked values and the wanted ones, I've created another PR to remove them, WDYT?
https://github.com/lemeurherve/updatecli/pull/18/files